src/render/tutorials/chrome-plugin/2_2.html.md

```diff
@@ -0,0 +1,495 @@
+
+Also, note the dashed transitions. They represent a transition that could potentially be triggered, but in this case wasn't.
+
+Finally, let's implement a node to handle the fallback case. The task should now look like this:
+
+```hocon
+payments-task {
+  action = fetch-user-info
+
+  onTransitions._success {
+    actions = [
+      {
+        action = fetch-payment-providers
+      },
+      {
+        action = fetch-offers-cb
+        
+        on._fallback {
+          action = fetch-offers-fallback
+        }
+      }
+    ]
+
+    onTransitions {
+      _success {
+        action = te-pebble
+      }
+      _error {
+        action = te-pebble
+      }
+    }
+  }
+}
+```
+
+We've added an action called `fetch-offers-fallback` that will be triggered when the circuit breaker returns a `_fallback` transition. This is a transition for the composite's subtask and not for the composite.
+
+The `fetch-offers-fallback` action could retrieve data from cache or from another service. In our case it'll just call the `/offers` endpoint:
+
+```hocon
+fetch-offers-fallback {
+  factory = http
+  config.endpointOptions {
+    path = /offers
+    domain = webapi
+    port = 8080
+  }
+}
+```
+
+When we refresh the page we still won't see any offers displayed on the page. We'll get to this in a minute. The graph should look like this:
+
+![Fallback](/img/blog/chrome-extension/with-fallback.png)
+
+As we can see, even though the circuit breaker ended with fallback, the composite as a whole ended with success because of the `fetch-offers-fallback` node.
+
+The only problem left is that we can't see any offers displayed, even though they were retrieved (you can check they were by viewing details of the fallback node). Let's take a look at the HTML snippet:
+
+```html
+<knotx:snippet data-knotx-task="payments-task">
+  <div>
+    <h2>Payment providers</h2>
+    <ul>
+      {% for provider in data['fetch-payment-providers']._result %}
+      <li>
+        <p>Payment provider: {{provider.name}}</p>
+        <p>Payment URL: {{provider.paymentUrl}}</p>
+      </li>
+      {% endfor %}
+    </ul>
+
+    <h2>Offers</h2>
+    <ul>
+      {% for offer in data['fetch-offers']._result %}
+      <li>
+        <p>Offer: {{offer.method}}</p>
+        <p>Price: {{offer.price}}</p>
+      </li>
+      {% endfor %}
+    </ul>
+  </div>
+</knotx:snippet>
+```
+
+Pebble retrieves offers data from a key called `fetch-offers`. By default, every action saves its data under its name. The problem is that in case of a timeout the action that retrieves data is called `fetch-offers-fallback` and therefore that's where the data is stored. One way to resolve this would be to use an action that rewrites data from one key to another. For the sake of this tutorial we'll just create additional markup for fallback messages. The Offers section of the snippet will look like this:

```

Comment:
```suggestion
Pebble retrieves offers data from a key called `fetch-offers`. By default, every action saves data under its name. The problem is that in case of a timeout the action that retrieves data is called `fetch-offers-fallback` and therefore that's where the data is stored. One way to resolve this would be to use an action that rewrites data from one key to another. For the sake of this tutorial, we'll just create additional markup for fallback messages. The Offers section of the snippet will look like this:
```